pkg/container-collection/options.go

```diff
@@ -146,6 +146,11 @@ func WithContainerRuntimeEnrichment(runtime *containerutils.RuntimeConfig) Conta
 			return containerRuntimeEnricher(runtime.Name, runtimeClient, container)
 		})
 
+		// skip enriching future containers for podman to avoid deadlocks
+		if runtime.Name == runtimeclient.PodmanName {
+			cc.containerEnrichers = nil
```

Comment:
I think you are removing all enrichers, including those unrelated to podman, such as WithTracerCollection(), WithLinuxNamespaceEnrichment().